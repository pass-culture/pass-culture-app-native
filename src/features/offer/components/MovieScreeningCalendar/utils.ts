import { OfferStockResponse } from 'api/gen'

export const getDateString = (date: string) => new Date(date).toDateString()

export const sortScreeningDates = (screeningDates: OfferStockResponse[]) =>
  [...screeningDates].sort((a, b) => {
    return a.beginningDatetime && b.beginningDatetime
      ? new Date(a.beginningDatetime).getTime() - new Date(b.beginningDatetime).getTime()
      : 0
  })
