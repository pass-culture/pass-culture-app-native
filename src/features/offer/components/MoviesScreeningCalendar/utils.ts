import { MOVIE_CALENDAR_PADDING } from 'features/offer/components/MovieCalendar/MovieCalendar'

export const handleMovieCalendarScroll = (
  currentIndex: number,
  flatListWidth: number,
  itemWidth: number
) => {
  const shift = flatListWidth / 2 - MOVIE_CALENDAR_PADDING
  const centerOfSelectedElement = currentIndex * itemWidth + itemWidth / 2

  if (centerOfSelectedElement - shift < 0 || currentIndex === 0) {
    return { offset: 0 }
  }
  return { offset: centerOfSelectedElement - shift }
}
