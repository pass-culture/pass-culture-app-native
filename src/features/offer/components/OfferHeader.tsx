import { t } from '@lingui/macro'
import { useRoute } from '@react-navigation/native'
import React, { useRef } from 'react'
import { Animated, Platform } from 'react-native'
import { useSafeAreaInsets } from 'react-native-safe-area-context'
import styled, { useTheme } from 'styled-components/native'

import { isApiError } from 'api/apiHelpers'
import { useAuthContext } from 'features/auth/AuthContext'
import {
  useAddFavorite,
  useFavorite,
  useRemoveFavorite,
} from 'features/favorites/pages/useFavorites'
import { UseRouteType } from 'features/navigation/RootNavigator'
import { getTabNavConfig } from 'features/navigation/TabBar/helpers'
import { useGoBack } from 'features/navigation/useGoBack'
import { SignUpSignInChoiceOfferModal } from 'features/offer/components/SignUpSignInChoiceOfferModal'
import { analytics } from 'libs/analytics'
import { WebShareModal } from 'libs/share/WebShareModal'
import { getAnimationState } from 'ui/components/headers/animationHelpers'
import { HeaderIcon } from 'ui/components/headers/HeaderIcon'
import { useModal } from 'ui/components/modals/useModal'
import { SNACK_BAR_TIME_OUT, useSnackBarContext } from 'ui/components/snackBar/SnackBarContext'
import { Spacer, Typo } from 'ui/theme'

import { useShareOffer } from '../services/useShareOffer'

interface Props {
  headerTransition: Animated.AnimatedInterpolation
  title: string
  offerId: number
}

/**
 * @param props.headerTransition should be between animated between 0 and 1
 */
export const OfferHeader: React.FC<Props> = (props) => {
  const { headerTransition, offerId, title } = props
  const { isLoggedIn } = useAuthContext()
  const { colors } = useTheme()
  const {
    visible: signInModalVisible,
    showModal: showSignInModal,
    hideModal: hideSignInModal,
  } = useModal(false)
  const {
    visible: shareOfferModalVisible,
    showModal: showShareOfferModal,
    hideModal: hideShareOfferModal,
  } = useModal(false)

  const { goBack } = useGoBack(...getTabNavConfig('Search'))
  const { share: shareOffer, shareContent } = useShareOffer(offerId)
  const { params } = useRoute<UseRouteType<'Offer'>>()
  const favorite = useFavorite({ offerId })
  const { showErrorSnackBar } = useSnackBarContext()
  const { top } = useSafeAreaInsets()

  const { mutate: addFavorite } = useAddFavorite({
    onSuccess: () => {
      if (typeof offerId === 'number') {
        const { from, moduleName } = params
        analytics.logHasAddedOfferToFavorites({ from, offerId, moduleName })
      }
    },
    onError: (error) => {
      showErrorSnackBar({
        message:
          isApiError(error) && error.content.code === 'MAX_FAVORITES_REACHED'
            ? t`Trop de favoris enregistrés. Supprime des favoris pour en ajouter de nouveaux.`
            : t`L'offre n'a pas été ajoutée à tes favoris`,
        timeout: SNACK_BAR_TIME_OUT,
      })
    },
  })

  const { mutate: removeFavorite } = useRemoveFavorite({
    onError: () => {
      showErrorSnackBar({
        message: t`L'offre n'a pas été retirée de tes favoris`,
        timeout: SNACK_BAR_TIME_OUT,
      })
    },
  })

  const { animationState, backgroundColor } = getAnimationState(headerTransition)
  const scaleFavoriteIconAnimatedValueRef = useRef(new Animated.Value(1))

  function pressFavorite() {
    if (!isLoggedIn) {
      showSignInModal()
    } else if (!favorite) {
      animateIcon(scaleFavoriteIconAnimatedValueRef.current)
      addFavorite({ offerId })
    } else if (favorite) {
      removeFavorite(favorite.id)
    }
  }

  const pressShareOffer = () => {
    shareOffer()
    showShareOfferModal()
  }

  return (
    <React.Fragment>
      <HeaderContainer style={{ backgroundColor }} safeAreaTop={top}>
        <Spacer.TopScreen />
        <Spacer.Column numberOfSpaces={2} />
        <Row>
          <Spacer.Row numberOfSpaces={6} />
          <HeaderIcon
            animationState={animationState}
            iconName="back"
            onPress={goBack}
            testID={t`Revenir en arrière`}
          />
          <Spacer.Row numberOfSpaces={3} />
          <Spacer.Row testID="leftShareIconPlaceholder" numberOfSpaces={10} />
          <Spacer.Flex />

          <Title testID="offerHeaderName" style={{ opacity: headerTransition }}>
            <Body>{title}</Body>
          </Title>

          <Spacer.Flex />
          <HeaderIcon
            animationState={animationState}
            iconName="share"
            onPress={pressShareOffer}
            testID={t`Partager`}
          />
          <Spacer.Row numberOfSpaces={3} />
          <HeaderIcon
            animationState={animationState}
            scaleAnimatedValue={scaleFavoriteIconAnimatedValueRef.current}
            initialColor={favorite ? colors.primary : undefined}
            iconName={favorite ? 'favorite-filled' : 'favorite'}
            onPress={pressFavorite}
            testID={t`Mettre en favoris`}
          />
          <Spacer.Row numberOfSpaces={6} />
        </Row>
        <Spacer.Column numberOfSpaces={2} />
      </HeaderContainer>
      {shareContent ? (
        <WebShareModal
          visible={shareOfferModalVisible}
          headerTitle={t`Partager l'offre`}
          shareContent={shareContent}
          dismissModal={hideShareOfferModal}
        />
      ) : null}
      <SignUpSignInChoiceOfferModal visible={signInModalVisible} dismissModal={hideSignInModal} />
    </React.Fragment>
  )
}

function animateIcon(animatedValue: Animated.Value): void {
  Animated.sequence([
    Animated.timing(animatedValue, {
      toValue: 1.3,
      duration: 200,
      useNativeDriver: false,
    }),
    Animated.timing(animatedValue, {
      toValue: 1,
      duration: 200,
      useNativeDriver: false,
    }),
  ]).start()
}

const HeaderContainer = styled(Animated.View)<{ safeAreaTop: number }>(
  ({ theme, safeAreaTop }) => ({
    position: 'absolute',
    top: 0,
    height: theme.appBarHeight + safeAreaTop,
    width: '100%',
    zIndex: theme.zIndex.header,
  })
)

const Row = styled.View({
  flex: 1,
  flexDirection: 'row',
  alignItems: 'center',
})
const Title = styled(Animated.Text).attrs({ numberOfLines: 2 })(({ theme }) => ({
  flexShrink: 1,
  textAlign: 'center',
  color: theme.colors.white,
  ...(Platform.OS === 'web' ? { whiteSpace: 'pre-wrap' } : {}),
}))

const Body = styled(Typo.Body)(({ theme }) => ({
  color: theme.colors.white,
}))
