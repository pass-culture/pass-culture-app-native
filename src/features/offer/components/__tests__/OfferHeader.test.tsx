import { NavigationContainer } from '@react-navigation/native'
import { act, cleanup, fireEvent, render, waitFor } from '@testing-library/react-native'
import { rest } from 'msw/'
import React from 'react'
import { Animated, Share } from 'react-native'
import waitForExpect from 'wait-for-expect'

import { OfferResponse } from 'api/gen'
import { useAuthContext } from 'features/auth/AuthContext'
import { Tab } from 'features/navigation/TabBar/TabNavigator'
import { offerResponseSnap } from 'features/offer/api/snaps/offerResponseSnap'
import { dehumanizeId } from 'features/offer/services/dehumanizeId'
import { analytics } from 'libs/analytics'
import { env } from 'libs/environment'
import { reactQueryProviderHOC } from 'tests/reactQueryProviderHOC'
import { server } from 'tests/server'
import { flushAllPromises } from 'tests/utils'

import { OfferHeader } from '../OfferHeader'

jest.mock('features/auth/AuthContext')
const mockUseAuthContext = useAuthContext as jest.Mock

jest.mock('@react-navigation/native', () => jest.requireActual('@react-navigation/native'))

describe('<OfferHeader />', () => {
  afterEach(async () => {
    jest.clearAllMocks()
    await cleanup()
  })
  it('should render correctly', async () => {
    const { toJSON } = await renderOfferHeader(true)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render all the icons - loggedIn', async () => {
    const offerHeader = await renderOfferHeader(true)
    expect(offerHeader.queryByTestId('icon-back')).toBeTruthy()
    expect(offerHeader.queryByTestId('icon-share')).toBeTruthy()
    expect(offerHeader.queryByTestId('icon-favorite')).toBeTruthy()
  })

  it('should render all the icons - not loggedIn', async () => {
    const offerHeader = await renderOfferHeader(false)
    expect(offerHeader.queryByTestId('icon-back')).toBeTruthy()
    expect(offerHeader.queryByTestId('icon-share')).toBeTruthy()
    expect(offerHeader.queryByTestId('icon-favorite')).toBeTruthy()
  })

  it('should fully display the title at the end of the animation', async () => {
    const { animatedValue, getByTestId } = await renderOfferHeader(true)
    expect(getByTestId('offerHeaderName').props.style.opacity).toBe(0)
    Animated.timing(animatedValue, { duration: 100, toValue: 1, useNativeDriver: false }).start()
    await waitForExpect(() => expect(getByTestId('offerHeaderName').props.style.opacity).toBe(1))
  })

  it('should call Share with the right arguments', async () => {
    const share = jest.spyOn(Share, 'share')
    const { getByTestId } = await renderOfferHeader(true)

    act(() => {
      fireEvent.press(getByTestId('icon-share'))
    })
    expect(share).toHaveBeenCalledTimes(1)
    const url = 'passculture://app.passculture.testing/offer/?id=116656'
    const title =
      'Retrouve Sous les Ã©toiles de Paris - VF chez PATHE BEAUGRENELLE sur le pass Culture'
    expect(share).toHaveBeenCalledWith({ message: title, title, url }, { dialogTitle: title })
  })

  describe('<OfferHeader /> - Analytics', () => {
    it('should log ShareOffer once when clicking on the Share button', async () => {
      const { getByTestId } = await renderOfferHeader(true)

      fireEvent.press(getByTestId('icon-share'))
      expect(analytics.logShareOffer).toHaveBeenCalledTimes(1)
      expect(analytics.logShareOffer).toHaveBeenCalledWith(offerId)

      fireEvent.press(getByTestId('icon-share'))
      fireEvent.press(getByTestId('icon-share'))
      expect(analytics.logShareOffer).toHaveBeenCalledTimes(1)
    })
  })
})

const humanizedOfferId = 'AHD3A'
// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
const offerId = dehumanizeId(humanizedOfferId)!

async function renderOfferHeader(isLoggedIn: boolean) {
  server.use(
    rest.get<OfferResponse>(env.API_BASE_URL + `/native/v1/offer/${offerId}`, (req, res, ctx) =>
      res(ctx.status(200), ctx.json(offerResponseSnap))
    )
  )
  mockUseAuthContext.mockImplementation(() => ({ isLoggedIn }))
  const animatedValue = new Animated.Value(0)
  const wrapper = render(
    reactQueryProviderHOC(
      <NavigationContainer>
        <Tab.Navigator initialRouteName="Home">
          <Tab.Screen name="Home">
            {() => (
              <OfferHeader
                title="Some very nice offer"
                headerTransition={animatedValue}
                offerId={offerId}
              />
            )}
          </Tab.Screen>
        </Tab.Navigator>
      </NavigationContainer>
    )
  )
  await act(async () => {
    await flushAllPromises()
  })
  await waitFor(() => wrapper.getByTestId('offerHeaderName'))
  return { ...wrapper, animatedValue }
}
