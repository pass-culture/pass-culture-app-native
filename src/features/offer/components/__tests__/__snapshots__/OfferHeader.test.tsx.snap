// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OfferHeader /> should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0)",
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <View
    customHeight={16}
    style={
      Array [
        Object {
          "height": 16,
        },
      ]
    }
  />
  <View
    numberOfSpaces={2}
    style={
      Array [
        Object {
          "height": 8,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      numberOfSpaces={6}
      style={
        Array [
          Object {
            "width": 24,
          },
        ]
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        animationState={
          Object {
            "iconBackgroundColor": "rgba(255, 255, 255, 1)",
            "iconBorderColor": "rgba(245, 245, 245, 1)",
            "transition": 0,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "borderColor": "rgba(245, 245, 245, 1)",
            "borderRadius": 40,
            "borderStyle": "solid",
            "borderWidth": 1,
            "justifyContent": "center",
            "width": 40,
          }
        }
        testID="headerIconRoundContainer"
      >
        <View
          size={32}
          style={
            Array [
              Object {
                "height": 32,
                "width": 32,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "opacity": 1,
                "position": "absolute",
              }
            }
            testID="initial-icon-container"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={32}
              bbWidth={32}
              focusable={false}
              height={32}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 32,
                    "width": 32,
                  },
                ]
              }
              testID="icon-back"
              vbHeight={32}
              vbWidth={32}
              width={32}
            >
              <RNSVGGroup>
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGGroup
                    fill={4279571733}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGPath
                          d="M8.94 19.466c-.301.277-.778.266-1.065-.025-.287-.29-.276-.75.026-1.027l5.314-4.878c1.746-1.603 4.49-1.603 6.237 0l5.314 4.878c.302.277.313.737.026 1.027-.287.29-.764.302-1.065.025l-5.315-4.878c-1.164-1.069-2.993-1.069-4.158 0L8.94 19.466z"
                          matrix={
                            Array [
                              -6.123233995736766e-17,
                              1,
                              1,
                              6.123233995736766e-17,
                              -0.3329999999999984,
                              -0.3329999999999984,
                            ]
                          }
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              Object {
                "opacity": 0,
                "position": "absolute",
              }
            }
            testID="final-icon-container"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={32}
              bbWidth={32}
              focusable={false}
              height={32}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 32,
                    "width": 32,
                  },
                ]
              }
              testID="final-icon-back"
              vbHeight={32}
              vbWidth={32}
              width={32}
            >
              <RNSVGGroup>
                <RNSVGGroup
                  fill={null}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                >
                  <RNSVGGroup
                    fill={4294967295}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGPath
                          d="M8.94 19.466c-.301.277-.778.266-1.065-.025-.287-.29-.276-.75.026-1.027l5.314-4.878c1.746-1.603 4.49-1.603 6.237 0l5.314 4.878c.302.277.313.737.026 1.027-.287.29-.764.302-1.065.025l-5.315-4.878c-1.164-1.069-2.993-1.069-4.158 0L8.94 19.466z"
                          matrix={
                            Array [
                              -6.123233995736766e-17,
                              1,
                              1,
                              6.123233995736766e-17,
                              -0.3329999999999984,
                              -0.3329999999999984,
                            ]
                          }
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
    <View
      numberOfSpaces={10}
      style={
        Array [
          Object {
            "width": 40,
          },
        ]
      }
      testID="headerIconPlaceholder"
    />
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    />
    <Text
      numberOfLines={2}
      style={
        Object {
          "flexShrink": 1,
          "opacity": 0,
          "textAlign": "center",
        }
      }
      testID="offerHeaderName"
    >
      <Text
        color="#ffffff"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 15,
              "lineHeight": 20,
            },
          ]
        }
      >
        Some very nice offer
      </Text>
    </Text>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        animationState={
          Object {
            "iconBackgroundColor": "rgba(255, 255, 255, 1)",
            "iconBorderColor": "rgba(245, 245, 245, 1)",
            "transition": 0,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "borderColor": "rgba(245, 245, 245, 1)",
            "borderRadius": 40,
            "borderStyle": "solid",
            "borderWidth": 1,
            "justifyContent": "center",
            "width": 40,
          }
        }
        testID="headerIconRoundContainer"
      >
        <View
          size={32}
          style={
            Array [
              Object {
                "height": 32,
                "width": 32,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "opacity": 1,
                "position": "absolute",
              }
            }
            testID="initial-icon-container"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={32}
              bbWidth={32}
              fill="#151515"
              focusable={false}
              height={32}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 32,
                    "width": 32,
                  },
                ]
              }
              testID="icon-share"
              vbHeight={40}
              vbWidth={40}
              width={32}
            >
              <RNSVGGroup
                fill={4279571733}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M15.773 15.2c.346 0 .625.28.625.624 0 .317-.235.578-.54.62l-.085.005h-3.481c-.54 0-.983.41-1.037.936l-.005.106v11.625c0 .54.41.983.935 1.036l.107.006h15c.539 0 .983-.41 1.036-.935l.005-.107V17.491c0-.539-.41-.983-.935-1.036l-.106-.006h-3.034c-.345 0-.625-.28-.625-.625 0-.316.235-.577.54-.619l.085-.006h3.034c1.218 0 2.215.952 2.287 2.153l.004.14v11.624c0 1.219-.951 2.215-2.152 2.288l-.14.004h-15c-1.218 0-2.215-.951-2.287-2.152l-.004-.14V17.491c0-1.218.951-2.215 2.152-2.287l.14-.005h3.481zm4.065-8.532l.029.003c.024.003.049.008.073.013l.018.005c.024.007.049.015.073.025l.025.011c.05.023.097.053.14.09l.007.006.059.06 4.651 5.315c.228.26.201.655-.058.882-.237.207-.584.204-.816.006l-.066-.065-3.557-4.064v14.71c0 .345-.28.625-.625.625-.316 0-.578-.236-.619-.54l-.006-.085V8.955l-3.556 4.064c-.227.26-.622.286-.882.059-.236-.207-.28-.552-.114-.808l.055-.074 4.652-5.316.028-.03.03-.029.008-.006c.02-.017.034-.027.049-.037l-.056.043c.028-.024.057-.046.088-.064.014-.01.03-.017.045-.025s.03-.014.046-.02l.028-.011c.021-.007.043-.014.065-.019l.052-.01.039-.003h.095z"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              Object {
                "opacity": 0,
                "position": "absolute",
              }
            }
            testID="final-icon-container"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={32}
              bbWidth={32}
              fill="#ffffff"
              focusable={false}
              height={32}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 32,
                    "width": 32,
                  },
                ]
              }
              testID="final-icon-share"
              vbHeight={40}
              vbWidth={40}
              width={32}
            >
              <RNSVGGroup
                fill={4294967295}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M15.773 15.2c.346 0 .625.28.625.624 0 .317-.235.578-.54.62l-.085.005h-3.481c-.54 0-.983.41-1.037.936l-.005.106v11.625c0 .54.41.983.935 1.036l.107.006h15c.539 0 .983-.41 1.036-.935l.005-.107V17.491c0-.539-.41-.983-.935-1.036l-.106-.006h-3.034c-.345 0-.625-.28-.625-.625 0-.316.235-.577.54-.619l.085-.006h3.034c1.218 0 2.215.952 2.287 2.153l.004.14v11.624c0 1.219-.951 2.215-2.152 2.288l-.14.004h-15c-1.218 0-2.215-.951-2.287-2.152l-.004-.14V17.491c0-1.218.951-2.215 2.152-2.287l.14-.005h3.481zm4.065-8.532l.029.003c.024.003.049.008.073.013l.018.005c.024.007.049.015.073.025l.025.011c.05.023.097.053.14.09l.007.006.059.06 4.651 5.315c.228.26.201.655-.058.882-.237.207-.584.204-.816.006l-.066-.065-3.557-4.064v14.71c0 .345-.28.625-.625.625-.316 0-.578-.236-.619-.54l-.006-.085V8.955l-3.556 4.064c-.227.26-.622.286-.882.059-.236-.207-.28-.552-.114-.808l.055-.074 4.652-5.316.028-.03.03-.029.008-.006c.02-.017.034-.027.049-.037l-.056.043c.028-.024.057-.046.088-.064.014-.01.03-.017.045-.025s.03-.014.046-.02l.028-.011c.021-.007.043-.014.065-.019l.052-.01.039-.003h.095z"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
    <View
      numberOfSpaces={3}
      style={
        Array [
          Object {
            "width": 12,
          },
        ]
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        animationState={
          Object {
            "iconBackgroundColor": "rgba(255, 255, 255, 1)",
            "iconBorderColor": "rgba(245, 245, 245, 1)",
            "transition": 0,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "aspectRatio": 1,
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "borderColor": "rgba(245, 245, 245, 1)",
            "borderRadius": 40,
            "borderStyle": "solid",
            "borderWidth": 1,
            "justifyContent": "center",
            "width": 40,
          }
        }
        testID="headerIconRoundContainer"
      >
        <View
          size={32}
          style={
            Array [
              Object {
                "height": 32,
                "width": 32,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "opacity": 1,
                "position": "absolute",
              }
            }
            testID="initial-icon-container"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={32}
              bbWidth={32}
              focusable={false}
              height={32}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 32,
                    "width": 32,
                  },
                ]
              }
              testID="icon-favorite"
              vbHeight={44}
              vbWidth={44}
              width={32}
            >
              <RNSVGGroup>
                <RNSVGDefs>
                  <RNSVGLinearGradient
                    gradient={
                      Array [
                        0,
                        -15395563,
                        1,
                        -15395563,
                      ]
                    }
                    gradientTransform={null}
                    gradientUnits={0}
                    name="prefix__a"
                    x1="-42.969%"
                    x2="153.672%"
                    y1="52.422%"
                    y2="52.422%"
                  />
                </RNSVGDefs>
                <RNSVGGroup
                  fill={null}
                  fillRule={0}
                  propList={
                    Array [
                      "fill",
                      "fillRule",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M19.703.558c3.784 0 6.852 3.137 6.852 7.008 0 .83-.142 1.642-.42 2.42a21.964 21.964 0 01-3.789 6.423.693.693 0 01-.992.072.732.732 0 01-.07-1.015 20.503 20.503 0 003.535-5.987c.217-.607.33-1.252.33-1.913 0-3.076-2.438-5.57-5.446-5.57-3.007 0-5.445 2.494-5.445 5.57 0 .959-1.406.959-1.406 0 0-3.077-2.437-5.57-5.444-5.57-3.008 0-5.446 2.494-5.446 5.57 0 .66.112 1.306.325 1.9 1.99 5.245 6.032 9.436 11.101 11.535l.163.065.284-.114a19.616 19.616 0 003.025-1.6l.478-.317a.692.692 0 01.975.196.729.729 0 01-.191.998 21.044 21.044 0 01-4.318 2.282.676.676 0 01-.39.032l-.096-.027-.053-.02C7.639 20.3 3.157 15.735.97 9.974a7.163 7.163 0 01-.415-2.408c0-3.87 3.068-7.008 6.853-7.008 2.648 0 4.946 1.537 6.086 3.787l.06.126.062-.126C14.724 2.16 16.924.646 19.477.562l.226-.004z"
                    fill={
                      Array [
                        1,
                        "prefix__a",
                      ]
                    }
                    fillRule={1}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        9,
                        10,
                      ]
                    }
                    propList={
                      Array [
                        "fill",
                        "fillRule",
                        "stroke",
                        "strokeWidth",
                      ]
                    }
                    stroke={
                      Array [
                        1,
                        "prefix__a",
                      ]
                    }
                    strokeWidth={0.5}
                  />
                  <RNSVGPath
                    d="M0 0h44v44H0z"
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <View
            style={
              Object {
                "opacity": 0,
                "position": "absolute",
              }
            }
            testID="final-icon-container"
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={32}
              bbWidth={32}
              focusable={false}
              height={32}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 32,
                    "width": 32,
                  },
                ]
              }
              testID="final-icon-favorite"
              vbHeight={44}
              vbWidth={44}
              width={32}
            >
              <RNSVGGroup>
                <RNSVGDefs>
                  <RNSVGLinearGradient
                    gradient={
                      Array [
                        0,
                        -1,
                        1,
                        -1,
                      ]
                    }
                    gradientTransform={null}
                    gradientUnits={0}
                    name="prefix__a"
                    x1="-42.969%"
                    x2="153.672%"
                    y1="52.422%"
                    y2="52.422%"
                  />
                </RNSVGDefs>
                <RNSVGGroup
                  fill={null}
                  fillRule={0}
                  propList={
                    Array [
                      "fill",
                      "fillRule",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M19.703.558c3.784 0 6.852 3.137 6.852 7.008 0 .83-.142 1.642-.42 2.42a21.964 21.964 0 01-3.789 6.423.693.693 0 01-.992.072.732.732 0 01-.07-1.015 20.503 20.503 0 003.535-5.987c.217-.607.33-1.252.33-1.913 0-3.076-2.438-5.57-5.446-5.57-3.007 0-5.445 2.494-5.445 5.57 0 .959-1.406.959-1.406 0 0-3.077-2.437-5.57-5.444-5.57-3.008 0-5.446 2.494-5.446 5.57 0 .66.112 1.306.325 1.9 1.99 5.245 6.032 9.436 11.101 11.535l.163.065.284-.114a19.616 19.616 0 003.025-1.6l.478-.317a.692.692 0 01.975.196.729.729 0 01-.191.998 21.044 21.044 0 01-4.318 2.282.676.676 0 01-.39.032l-.096-.027-.053-.02C7.639 20.3 3.157 15.735.97 9.974a7.163 7.163 0 01-.415-2.408c0-3.87 3.068-7.008 6.853-7.008 2.648 0 4.946 1.537 6.086 3.787l.06.126.062-.126C14.724 2.16 16.924.646 19.477.562l.226-.004z"
                    fill={
                      Array [
                        1,
                        "prefix__a",
                      ]
                    }
                    fillRule={1}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        9,
                        10,
                      ]
                    }
                    propList={
                      Array [
                        "fill",
                        "fillRule",
                        "stroke",
                        "strokeWidth",
                      ]
                    }
                    stroke={
                      Array [
                        1,
                        "prefix__a",
                      ]
                    }
                    strokeWidth={0.5}
                  />
                  <RNSVGPath
                    d="M0 0h44v44H0z"
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
      </View>
    </View>
    <View
      numberOfSpaces={6}
      style={
        Array [
          Object {
            "width": 24,
          },
        ]
      }
    />
  </View>
  <View
    numberOfSpaces={2}
    style={
      Array [
        Object {
          "height": 8,
        },
      ]
    }
  />
</View>
`;
